libs/mockito-extensions/src/main/java/org/codice/ddf/test/mockito/PrivilegedVerificationMode.java

```diff
@@ -0,0 +1,143 @@
+import org.mockito.internal.verification.VerificationModeFactory;
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.internal.verification.api.VerificationDataInOrder;
+import org.mockito.internal.verification.api.VerificationInOrderMode;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.Location;
+import org.mockito.invocation.MatchableInvocation;
+import org.mockito.verification.VerificationMode;
+
+/**
+ * Class that extends Mockito's {@link VerificationMode} to verify that a mocked method has been
+ * called within a {@code AccessController#doPrivileged()} block.
+ *
+ * <p><b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+@Incubating
+public class PrivilegedVerificationMode implements VerificationMode, VerificationInOrderMode {
+
+  private final VerificationMode mode;
+
+  private PrivilegedVerificationMode(VerificationMode mode) {
+    this.mode = mode;
+  }
+
+  /**
+   * Wraps another {@link VerificationMode} instance and ensures all {@code verify()} calls from the
+   * specified mode were made inside a {@code AccessController.doPrivileged()} block.
+   *
+   * @param mode {@link VerificationMode} instance to wrap with a privileged check
+   * @return wrapped {@link VerificationMode} instance
+   */
+  public static VerificationMode privileged(VerificationMode mode) {
+    return new PrivilegedVerificationMode(mode);
+  }
+
+  /**
+   * Returns a {@link VerificationMode} object that ensures a single {@code verify} code was made
+   * inside a {@code AccessController.doPrivileged()} block.
+   *
+   * @return wrapped {@link VerificationMode} instance
+   */
+  public static VerificationMode privileged() {
+    return new PrivilegedVerificationMode(times(1));
+  }
+
+  @Override
+  public void verify(VerificationData verificationData) {
+    mode.verify(verificationData);
+
+    verify(verificationData.getTarget(), verificationData.getAllInvocations());
+  }
+
+  @Override
+  public void verifyInOrder(VerificationDataInOrder verificationDataInOrder) {
+    if (!(mode instanceof VerificationInOrderMode)) {
+      throw new MockitoAssertionError(mode.toString() + " cannot be used in ordered mode");
+    }
+
+    ((VerificationInOrderMode) mode).verifyInOrder(verificationDataInOrder);
+
+    verify(verificationDataInOrder.getWanted(), verificationDataInOrder.getAllInvocations());
+  }
+
+  @Override
+  public VerificationMode description(String description) {
+    return VerificationModeFactory.description(this, description);
+  }
+
+  private void verify(MatchableInvocation wanted, List<Invocation> invocations) {
+
+    if (invocations.isEmpty()) {
+      return;
+    }
+
+    try {
+      Location location = invocations.get(0).getLocation();
+      Field stackTraceHolderField = getStackTraceHolderField(location);
+
+      for (Invocation invocation : invocations) {
```

Comment:
✏️ This is extremely pedantic, but 
```java
invocations
  .stream()
  .filter()
```
would read a little cleaner to me. You'd want to extract the catch clause to a named function, but one could argue that also improves readability.